File: exercism.io/javascript/series/series.js
//
// This is only a SKELETON file for the 'Series' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Series {
  constructor(numbers) {
    this.numbers = numbers
  }

  get digits() {
    let result = []
    for(let i = 0 ;i < this.numbers.length; i++) result.push(parseInt(this.numbers[i]))
    return result
  }

  slices(num) {
    if(num > this.digits.length) throw new Error("Slice size is too big.")
    let result = []
    for(let i = 0; i <= this.digits.length - num; i++){
      result.push(this.digits.slice(i, i + num))
    }
    return result
  }
}

File: exercism.io/python/series/series.py
def slices(series, length):

    l = len(series)
    if length > l or length <= 0:
        raise ValueError(".+")

    result = []
    i = 0
    while(i <= l - length):
        result.append(series[i:i+length])
        i +=1

    return result
